GVRf/Framework/src/org/gearvrf/GVRMesh.java

```diff
@@ -66,8 +67,7 @@ protected final boolean registerWrapper() {
      *            Array containing the packed vertex data.
      */
     public void setVertices(float[] vertices) {
-        isNonNull(vertices);
-        isValidData(vertices.length, 3);
+        validOtherCheck(INPUT_TYPE_VERTICES, vertices, vertices.length, 3);
```

Comment:
Actually I am also quite sure about this part. Because there are general two types of check operations. One is for all kinds of vector, the other one is for the data other than vector.

For INPUT_TYPE_VERTICES, imho, I am thinking that there are two types of design to manage those constant String, one is let Preconditions class hold all those Strings. The pros is that it is easy to manage and extend, the cons is that actually as you say, it is going to put something that doesn't belong to Preconditions and it may disobey the concept of OOD. 

The second way is to let the class itself to hold those constant String. The pros is that we are going to put something to where it should be. But the cons, in my opinion, is that there may cause some duplicates. For example, INPUT_TYPE_VECTOR actually applies to many classes not only limited to GVRMesh, if we put it at every class it may be used, it would be a little redundant.

Jon, How do you think?